### ISSUE TITLE
Input logits shape is torch.Size([1, 141]), but vocabulary is size 31. Need logits of shape: (time, vocabulary)

### ISSUE THREAD

**Issue by GIN-Margarita** [idiap/w2v2-air-traffic#8]
![image](https://github.com/user-attachments/assets/96afa29c-0fee-4a72-b4c6-83c32fbd036c)

when I try to use the pretrained Language_model to inference, i  face this problem,please !

**Comment by GIN-Margarita** [comment]
I fixed this ! update the eval_model.py



def get_kenlm_processor(model_path, path_lm=None):
    """Function that instantiate the models and then gives them back for evaluation"""

    path_tokenizer = model_path

    # check that we send the right dir where the model is stored
    if Path(model_path).is_dir():
        processor = Wav2Vec2Processor.from_pretrained(model_path)
        model = AutoModelForCTC.from_pretrained(model_path)
    else:
        print(f"Error. Models were not found in {model_path}")

    # In case we don't pass any language model path, we just send back the model and processor
    if path_lm is None:
        return processor, None, None, model

    # vocab = processor.tokenizer.convert_ids_to_tokens(
    #     range(0, processor.tokenizer.vocab_size)
    # )
    # we need to add these tokens in the tokenizer
    # vocab.append("<s>")
    # vocab.append("</s>")

    # instantiate the tokenizer
    # tokenizer = Wav2Vec2CTCTokenizer(
    #     path_tokenizer + "/vocab.json",
    #     unk_token="[UNK]",
    #     pad_token="[PAD]",
    #     word_delimiter_token="|",
    # )

    # load CTC decoder WITH a LM and HuggingFace processor with CTC decoder and LM
    # ctcdecoder_kenlm = build_ctcdecoder(
    #     labels=vocab,
    #     kenlm_model_path=path_lm,
    # )
    # HuggingFace processor with CTC decoder (with LM)
    processor_ctc_kenlm = Wav2Vec2ProcessorWithLM.from_pretrained(model_path)
    # processor_ctc_kenlm = Wav2Vec2ProcessorWithLM(
    #     feature_extractor=processor.feature_extractor,
    #     tokenizer=tokenizer,
    #     decoder=ctcdecoder_kenlm,
    # )

    return processor, processor_ctc_kenlm, model
    
    update the function to finish eval!
